doc/developer-guides/coding_guidelines.rst

```diff
@@ -2787,3 +2722,418 @@ Compliant example::
+
+   Non-compliant example::
+
+       if (condition) {
+         do_a();
+       } else {
+         do_b();
+       }
+
+
+IT-02: Tabs shall be 8 characters wide
+--------------------------------------
+
+A tab character shall be considered 8-character wide when limiting the line
+width.
+
+Compliant example::
+
+    /* the following line hits the line limit (120 characters) if tab is counted as 8-character wide */
+            /*1---------2---------3---------4---------5---------6---------7---------8---------9---------A---------B-------*/
+
+.. rst-class:: non-compliant-code
+
+   Non-compliant example::
+
+       /* the following line hits the line limit (120 characters) if tab is counted as 4-character wide */
+           /*1---------2---------3---------4---------5---------6---------7---------8---------9---------A---------B-------$$$$*/
+
+
+
+Whitespaces
+===========
+
+WS-01: Trailing whitespaces shall not be allowed at the end of lines
+--------------------------------------------------------------------
+
+Compliant example::
+
+    uint32_t a;
+    uint32_t b;
+    uint32_t c;
+
+.. rst-class:: non-compliant-code
+
+   Non-compliant example::
+
+       uint32_t a;    
+       uint32_t b;    
+       uint32_t c;    
+       
+
+
+WS-02: A single space shall exist between non-function-like keywords and opening brackets
+-----------------------------------------------------------------------------------------
+
+A single space shall exist between a non-function-like keyword and the opening
+bracket (either a brace or a parenthesis) that follows. This rule applies to the
+keywords `if`, `else`, `for`, `do`, `while`, `switch` and `return`. This rule
+does not apply to the keyword `sizeof`.
+
+Compliant example::
+
+    uint32_t showcase;
+    
+    if (showcase == 0U) {
+        showcase = 32U;
+    }
+
+.. rst-class:: non-compliant-code
+
+   Non-compliant example::
+
+       uint32_t showcase;
+       
+       if(showcase == 0U){
+           showcase = 32U;
+       }
+
+
+WS-03: Space shall not exist between the function identifier and the following open-parenthesis
+-----------------------------------------------------------------------------------------------
+
+Compliant example::
+
+    size_t entry_size = sizeof(struct vcpuid_entry);
+
+.. rst-class:: non-compliant-code
+
+   Non-compliant example::
+
+       size_t entry_size = sizeof (struct vcpuid_entry);
+
+
+WS-04: Space shall not exist right after opening brackets and right before closing ones
+---------------------------------------------------------------------------------------
+
+Brackets in this rule refer to parenthesis, braces and squared brackets.
+
+Compliant example::
+
+    size_t entry_size = sizeof(struct vcpuid_entry);
+
+.. rst-class:: non-compliant-code
+
+   Non-compliant example::
+
+       size_t entry_size = sizeof( struct vcpuid_entry );
+
+
+WS-05: The `*` characters used for pointers shall be right before the function or variable identifiers
```

Comment:
Please replace all uses of the back tick `` ` `` with an apostrophe `` ' ``

Reply:
Will update it in v3.